common/src/main/java/com/wynntils/wc/utils/lootrun/LootrunUtils.java

```diff
@@ -0,0 +1,544 @@
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.wc.utils.lootrun.objects.*;
+import it.unimi.dsi.fastutil.Pair;
+import it.unimi.dsi.fastutil.longs.Long2ObjectMap;
+import it.unimi.dsi.fastutil.longs.Long2ObjectOpenHashMap;
+import it.unimi.dsi.fastutil.objects.ObjectObjectImmutablePair;
+import java.io.*;
+import java.text.DateFormat;
+import java.util.*;
+import javax.annotation.Nullable;
+import net.minecraft.ChatFormatting;
+import net.minecraft.client.Camera;
+import net.minecraft.client.gui.Font;
+import net.minecraft.client.gui.screens.inventory.ContainerScreen;
+import net.minecraft.client.player.LocalPlayer;
+import net.minecraft.client.renderer.LevelRenderer;
+import net.minecraft.client.renderer.MultiBufferSource;
+import net.minecraft.client.renderer.RenderType;
+import net.minecraft.commands.CommandSourceStack;
+import net.minecraft.core.BlockPos;
+import net.minecraft.network.chat.*;
+import net.minecraft.network.chat.Component;
+import net.minecraft.util.CubicSpline;
+import net.minecraft.util.FastColor;
+import net.minecraft.util.FormattedCharSequence;
+import net.minecraft.util.Mth;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.level.ChunkPos;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.Blocks;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.phys.AABB;
+import net.minecraft.world.phys.Vec3;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class LootrunUtils {
+
+    public static final File LOOTRUNS = new File(McUtils.mc().gameDirectory, "lootruns");
+
+    private static final List<Integer> COLORS = List.of(
+            ChatFormatting.RED.getColor(),
+            ChatFormatting.GOLD.getColor(),
+            ChatFormatting.YELLOW.getColor(),
+            ChatFormatting.GREEN.getColor(),
+            ChatFormatting.BLUE.getColor(),
+            0x3f00ff,
+            ChatFormatting.DARK_PURPLE.getColor());
+
+    private static final int ACTIVE_COLOR = ChatFormatting.AQUA.getColor();
+    private static final int RECORDING_COLOR = 0xff0000;
+
+    public static LootrunInstance lootrun = null;
+    public static LootrunUncompiled uncompiled = null;
+    public static LootrunInstance recordingCompiled = null;
+    public static LootrunUncompiled recording = null;
+
+    public static RecordingInformation recordingInformation = null;
+
+    public static void register() {
+        WynntilsMod.getEventBus().register(LootrunUtils.class);
+    }
+
+    public static void unregister() {
+        WynntilsMod.getEventBus().register(LootrunUtils.class);
+    }
+
+    @SubscribeEvent
+    public static void render(RenderLevelLastEvent event) {
+        renderLootrun(event, lootrun, ACTIVE_COLOR);
+        renderLootrun(event, recordingCompiled, RECORDING_COLOR);
+    }
+
+    private static void renderLootrun(RenderLevelLastEvent event, LootrunInstance lootrun, int color) {
```

Comment:
The rendering for lootrun paths seems kind of broken - sometimes it'll stop rendering a certain portion when I move around
![image](https://user-images.githubusercontent.com/57310593/167705574-076657a9-4a07-4f1d-a056-f7d5f60a5750.png)
![image](https://user-images.githubusercontent.com/57310593/167705612-8333f369-b426-4d3c-9eb5-a3601e3f12e1.png)
Unless this is intentional and you only want to load portions closest to the player?

Reply:
I am not sure about this, I will check later but maybe coehlrich knows better

Reply:
Also, I can't seem to reproduce this, can you maybe share the affected file here?

Reply:
@DonkeyBlaster you had a low render distance when this happened? If so, it might be related to valid block calculation math. If not, I have no idea.

Reply:
Yeah, my laptop has a render setting of 2 chunks

Reply:
I checked the code and render distance should not affect this.

Reply:
Fixed this